kratos/includes/indexed_object.h

```diff
@@ -96,6 +96,18 @@ class IndexedObject
         return rThisObject.Id();
     }
 
+    template<class TObjectType>
```

Comment:
templating this is a bit dangerous. don't we want it to work only for the same type?

Reply:
I was following the signature of the `operator()` to be consistent. I was not sure why `operator()` was templated. @pooyan-dadvand ?